test/vaults/vaults.test.ts

```diff
@@ -0,0 +1,586 @@
+import { BigNumber, constants } from "ethers";
+import ProofsFetcher from "../../src/vaults/ProofsFetcher";
+import { getAllProofs } from "../../src/utils/getCurrentOnChainDistribution";
+import { parseUnits } from "ethers/lib/utils";
+import { distributorFromEvents } from "./utils";
+import { VaultEventType } from "../../src/vaults/Distributor";
+import { allEpochs, EpochConfig } from "../../src";
+
+describe("Vaults Distributor", () => {
+  const vaultAddress = "0x6abfd6139c7c3cc270ee2ce132e309f59caaf6a2";
+  const proofsFetcher = new ProofsFetcher();
+  const epochConfig = proofsFetcher.getEpochFromId("age1-epoch1");
+  const allProofs = getAllProofs();
+
+  it("Should distribute to the Deposit owner", async () => {
+    const distributor = distributorFromEvents(vaultAddress, [
+      {
+        type: VaultEventType.Deposit,
+        event: {
+          blockNumber: epochConfig.initialBlock!,
+          transactionIndex: 1,
+          logIndex: 1,
+          args: {
+            caller: constants.AddressZero,
+            owner: "0x0000000000000000000000000000000000000001",
+            assets: parseUnits("1000"),
+            shares: parseUnits("1000"),
+          },
+        },
+      },
+    ]);
+    const { lastMerkleTree: merkleTree } = await distributor.distributeMorpho(epochConfig.id);
+    expect(merkleTree.proofs["0x0000000000000000000000000000000000000001"]?.amount).toBeDefined();
+    expect(merkleTree.proofs["0x0000000000000000000000000000000000000000"]?.amount).toBeUndefined();
+  });
+  it("Should distribute to the Withdrawer owner", async () => {
+    const distributor = distributorFromEvents(vaultAddress, [
+      {
+        type: VaultEventType.Deposit,
+        event: {
+          blockNumber: epochConfig.initialBlock!,
+          transactionIndex: 1,
+          logIndex: 1,
+          args: {
+            caller: constants.AddressZero,
+            owner: "0x0000000000000000000000000000000000000001",
+            assets: parseUnits("1000"),
+            shares: parseUnits("1000"),
+          },
+        },
+      },
+      {
+        type: VaultEventType.Deposit,
+        event: {
+          blockNumber: epochConfig.initialBlock!,
+          transactionIndex: 1,
+          logIndex: 1,
+          args: {
+            caller: constants.AddressZero,
+            owner: "0x0000000000000000000000000000000000000010",
+            assets: parseUnits("1000"),
+            shares: parseUnits("1000"),
+          },
+        },
+      },
+      {
+        type: VaultEventType.Withdraw,
+        event: {
+          blockNumber: epochConfig.initialBlock! + 1000,
+          transactionIndex: 1,
+          logIndex: 1,
+          args: {
+            owner: "0x0000000000000000000000000000000000000001",
+            receiver: "0x0000000000000000000000000000000000000002",
+            caller: "0x0000000000000000000000000000000000000003",
```

Comment:
perhaps easier to read if you set those variables as constants like user1, user2, ...